qnap/shared/teliod.cgi

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+echo "Content-Type: application/json"
+echo ""
+
+# Varibles set by the HTTP web server
+#
+# $REQUEST_METHOD
+# $QUERY_STRING
+# $CONTENT_LENGTH
+
+POST_DATA=""
+
+if [[ "$REQUEST_METHOD" == "POST" || "$REQUEST_METHOD" == "PATCH" ]]; then
+    read -n $CONTENT_LENGTH POST_DATA
+fi
+
+QPKG_DIR="/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet"
+APP_TMP_DIR="/tmp/nordsecuritymeshnet"
+APP_CMD="$QPKG_DIR/teliod"
+CONFIG_FILE="$QPKG_DIR/teliod.cfg"
+LOG_FILE="$APP_TMP_DIR/teliod.log"
+PID_FILE="$APP_TMP_DIR/teliod.pid"
+
+is_application_running() {
+    if [ -f ${PID_FILE} ]; then
+        PID=$(cat ${PID_FILE})
+        if [ -d /proc/${PID}/ ]; then
+            echo 1
+        fi
+    fi
+}
+
+send_response() {
+    local code=$1
+    local message=$2
+    echo "{\"code\": $code, \"message\": \"$message\"}"
+    exit $code
+}
+
+start_application() {
+    if [ is_application_running -eq 1 ]; then
+            send_response 400 "Application is already running."
+    fi
+    $APP_CMD daemon $CONFIG_FILE &
+    if [[ $? -eq 0 ]]; then
+        echo $! > ${PID_FILE}
```

Comment:
nit: `echo $! > ${PID_FILE}` maybe move this before if statement, since we are doing same thing in both cases.

Reply:
in that case `$?` would return the exit code of that echo command instead of the `$APP_CMD`

Reply:
oh yeah, true, missed that